### PR TITLE
Subscriber can comment on article

### PR DESCRIPTION
**Pull request by johannaanden**
## Feature
- Creates comment model
- Creates comment controller
- Adds comment form
- Adds feature and unit tests

## Gem
- Faker

## Pivotal tracker
[PT Link](https://www.pivotaltracker.com/story/show/163088250)

## Screenshot
screenshot 2019-01-17 at 15 55 55

### CODE REVIEW

File: app/views/articles/show.html.haml [hunk 1]
```diff
@@ -5,4 +5,28 @@
...
+        %h2 Comments
+        - @article.comments.each do |comment|
+            %p
+                %strong Commenter:
+                = comment.user.name
+            %p
+                %strong Comment:
+                = comment.body
+
+        - if user_signed_in?
```
Comment (oliverochman) [inline app/views/articles/show.html.haml]: The feature name and user story says that only subscribers should be able to comment

File: features/subscriber_can_comment_on_article.feature [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            | Bill    | bill@mail.com     | password | member |
+            | Johanna | johanna@email.com | password | member |
+
+        And the following article exists:
+            | title | description    | content               | user_id | category_id | published |
+            | News  | This is a news | I like to eat cheese! | Bill    | Weather     | true      |
+
+        And I visit the landing page
+
+    Scenario: Logged in user can submit a comment [Happy Path]
```
Comment (Jon1969Edwards) [inline features/subscriber_can_comment_on_article.feature]: Should this be said in first person IE: 'When I am logged in...'?

File: db/schema.rb [hunk 3]
```diff
@@ -25,6 +25,7 @@
     t.bigint "category_id"
     t.bigint "user_id"
     t.boolean "published", default: false
+    t.boolean "premium", default: false
```
Comment (oliverochman) [inline db/schema.rb]: why was this added in this PR

File: features/subscriber_can_comment_on_article.feature [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        Given I am logged in as 'johanna@email.com'
+        And I click on 'News'
+        And I fill in 'Body' field with 'This is my comment'
+        And I click on 'Create Comment'
+        Then I should see 'Johanna'
+        And I should see 'This is my comment'
+
+    Scenario: Visitor cannot submit a comment [Sad Path]
```
Comment (Jon1969Edwards) [inline features/subscriber_can_comment_on_article.feature]: Same here with first person?

File: features/subscriber_can_comment_on_article.feature [hunk 5]
```diff
@@ -0,0 +1,30 @@
+Feature: Subscriber can comment on article
+
+    As a Subscriber
+    In order to voice my opinion or ask questions
+    I would like to be able to leave a comment on an article
+
+    Background:
+        Given the following user exists
+            | name    | email             | password | role   |
+            | Bill    | bill@mail.com     | password | member |
```
Comment (oliverochman) [inline features/subscriber_can_comment_on_article.feature]: Shouldn't Bill be a journalist?

File: spec/models/comment_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'DB table' do
+    it { is_expected.to have_db_column :body }
+  end
+
```
Comment (oliverochman) [inline spec/models/comment_spec.rb:7]: you need to test the associations here

you also need to test the associations in the other unit tests as well

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/subscriber_can_comment_on_article.feature (hunk 2 above)
Comment (CamronLDNF) [reply features/subscriber_can_comment_on_article.feature]: This is a Scenario description and not a step. But agree, will add an "A" to make it more readable.

File: features/subscriber_can_comment_on_article.feature (hunk 4 above)
Comment (CamronLDNF) [reply features/subscriber_can_comment_on_article.feature]: Same as above answer. Added an "A".

File: app/views/articles/show.html.haml (hunk 1 above)
Comment (CamronLDNF) [reply app/views/articles/show.html.haml]: Indeed it does! Corrected the code now. Thanks.

File: db/schema.rb (hunk 3 above)
Comment (CamronLDNF) [reply db/schema.rb]: Removed now.

File: features/subscriber_can_comment_on_article.feature (hunk 5 above)
Comment (CamronLDNF) [reply features/subscriber_can_comment_on_article.feature]: True. Changed now.

File: spec/models/comment_spec.rb (hunk 6 above)
Comment (CamronLDNF) [reply spec/models/comment_spec.rb:7]: Done!

**Review by oliverochman — APPROVED** [review]
LGTM!
